### ISSUE TITLE
Broken posts without --branch?

### ISSUE THREAD

**Issue by cryptix** [ssbc/patchwork#166]
I think I broke something when I started a script which posted with `--root` but without `--branch`.

Root post: `%iZk0Xcf3mZ1i4ZUJSYMU6/1LJgU4vP5IXSFUt5L9Vcc=.sha256`

Clicking on the replies link gives me an empty view:

![ssb-empty](https://cloud.githubusercontent.com/assets/111202/11686597/094d3c42-9e82-11e5-8694-a397ccf1a190.png)

And this unexpected error in the warning icon:

```
Cannot read property 'link' of null

TypeError: Cannot read property 'link' of null
    at insertReply (/home/cryptix/patchwork/node_modules/patchwork-threads/index.js:82:46)
    at flattenAndReorderReplies (/home/cryptix/patchwork/node_modules/patchwork-threads/index.js:113:7)
    at Array.forEach (native)
    at Object.exports.flattenThread (/home/cryptix/patchwork/node_modules/patchwork-threads/index.js:72:11)
    at /home/cryptix/patchwork/ui/com/msg-thread.jsx:56:25
    at /home/cryptix/patchwork/node_modules/patchwork-threads/index.js:231:7
    at /home/cryptix/patchwork/node_modules/multicb/index.js:40:13
    at /home/cryptix/patchwork/node_modules/patchwork-threads/index.js:258:25
    at /home/cryptix/patchwork/node_modules/multicb/index.js:40:13
    at /home/cryptix/patchwork/node_modules/patchwork-threads/index.js:299:7

This was an unhandled exception.
```

Going back to the feed/newspaper icon, it's also just empty and I need to restart.

![ssb-empty-feed](https://cloud.githubusercontent.com/assets/111202/11686607/1e0cedd0-9e82-11e5-8bfa-0cdc1ccd5319.png)

**Comment by dominictarr** [comment]
This in an interesting case - people are gonna post malformed messages - how should we deal with it?

@cryptix we have a library to create the valid post types, easiest to use that to make sure the post is consistent: https://github.com/ssbc/ssb-msg-schemas#type-post

**Comment by pfraze** [comment]
We have to treat all messages as untrusted input. It's the same techniques as for incoming web requests. 

In this case we can just assume, if there's a root but no branch, that the branch should be root. So, it's a recoverable error.

**Comment by pfraze** [comment]
[Fixed with patchwork-threads@1.1.2](https://github.com/ssbc/patchwork-threads/commit/2352412574bd1780120953dc2f5dcf4ee4ae4234)
